test/sanity/validate-modules/validate-modules

```diff
@@ -782,9 +892,12 @@ def main():
                 with ModuleValidator(path, analyze_arg_spec=args.arg_spec,
                                      base_branch=args.base_branch) as mv:
                     mv.validate()
-                    exit.append(mv.report(args.warnings))
+                    reports.update(mv.report())
 
-    sys.exit(sum(exit))
+    if args.format == 'plain':
+        sys.exit(Reporter.plain(reports, warnings=args.warnings, output=args.output))
+    else:
+        sys.exit(Reporter.json(reports, warnings=args.warnings, output=args.output))
```

Comment:
That works. Just so we can look at the exit code to know whether or not the program exited "normally", which in this case means writing out a valid JSON report. The `shellcheck` program does something similar. From the `shellcheck` man page:

```
ShellCheck uses the follow exit codes:

       o 0: All files successfully scanned with no issues.
       o 1: All files successfully scanned with some issues.
       o 2: Some files could not be processed (e.g.  file not found).
       o 3: ShellCheck was invoked with bad syntax (e.g.  unknown flag).
       o 4: ShellCheck was invoked with bad options (e.g.  unknown formatter).
```